### PR TITLE
Recursively convert images and don't write to source directory

### PR DESCRIPTION
**Pull request by emmmile**
I have this proposal, I'm happy to discuss it otherwise I'll keep working on my fork! :)

- I thought was handy to recursively convert images.
- it's probably not a good idea to write in the source folder. My scenario is that I have some png files (original pictures) in `/assets/something` that I want to serve as wepb from `_site/assets/something`. But I don't want to have the original png files in the `_site` directory, so this kind of fixes that.

### CODE REVIEW

File: lib/jekyll-webp/webpGenerator.rb [hunk 1]
```diff
@@ -44,57 +45,55 @@ def generate(site)
...
-
-              # Generate the file
-              WebpExec.run(@config['quality'], infile_fullpath, outfile_fullpath_webp)
-              
-              # Keep the webp file from being cleaned by Jekyll
-              site.static_files << WebpFile.new(site, site.dest, imgdir, outfile_filename)
-              file_count += 1
-              
-          end # dir.foreach
+        for source in @config['img_dir']
```
Comment (emmmile) [inline lib/jekyll-webp/webpGenerator.rb]: I changed the indentation to match the rest of the file.

Comment (sverrirs) [reply lib/jekyll-webp/webpGenerator.rb]: Please avoid this kind of non-code changes as it makes it really difficult to review the changes you made.

File: lib/jekyll-webp/webpExec.rb [hunk 2]
```diff
@@ -27,8 +27,8 @@ def self.run(quality, input_file, output_file)
         full_path = File.join(gem_root, bin_path, exe_name)
 
         # Construct the full program call
-        cmd = "\"#{full_path}\" -quiet -mt -q #{quality.to_s} \"#{input_file}\" -o \"#{output_file}\""
-        
+        cmd = "\"#{full_path}\" -quiet -mt -m 6 -pass 10 -q #{quality.to_s} \"#{input_file}\" -o \"#{output_file}\""
```
Comment (sverrirs) [inline lib/jekyll-webp/webpExec.rb]: Please expose these controls as configurable values to enable keeping existing behaviour.

File: lib/jekyll-webp/webpGenerator.rb [hunk 3]
```diff
@@ -30,11 +31,11 @@ def generate(site)
 
         # If disabled then simply quit
         if !@config['enabled']
-          Jekyll.logger.info "WebP:","Disabled in site.config."
+          Jekyll.logger.info "WebP:", "Disabled in site.config."
```
Comment (sverrirs) [inline lib/jekyll-webp/webpGenerator.rb]: Please avoid non-code, style changes like this addition of spaces.

File: lib/jekyll-webp/webpGenerator.rb (hunk 1 above)
Comment (sverrirs) [inline lib/jekyll-webp/webpGenerator.rb]: this local variable was intentionally coded as `imgdir` to avoid confusing it with the site.source

**Comment by sverrirs** [comment]
Thank you for providing this PR it is appreciated that you took the time to enhance the functionality :)

While the addition you made is a valuable feature there are too many non-feature related changes you made to the code for me to accept the PR in this state. 

Could I please ask you to rework your change to achieve the same functionality without:
- Changing indent levels or applying personal style changes such as space and line breaks.
- Renaming of local variables.

:)

**Comment by emmmile** [comment]
Thank you for reviewing so quickly! 

I am going revert the changes you mentioned (some of them actually were automatically made by my editor) keeping the functionality. Instead of variable name changes I'll add some comments mabye if there is confusion.

**Comment by emmmile** [comment]
Ok, this is it. I think will do the compression changes in another PR :)

**Review by sverrirs — APPROVED** [review]
Excellent, thank you very much for this feature and for your patience with my earlier comments :)

**Comment by sverrirs** [comment]
Also, feel free to bump the version to 1.0 in your next commit.

**Comment by emmmile** [comment]
Okay! Another thing, I forgot to tag the new branch. Or it's something you want to do on `master`?

**Comment by sverrirs** [comment]
I'll tag it on master no problem

**Pull request merged**